packages/kbn-test/src/functional_test_runner/lib/suite_tracker.test.ts

```diff
@@ -0,0 +1,197 @@
+const ENVS_TO_RESET = ['TEST_METADATA_PATH'];
+
+describe('SuiteTracker', () => {
+  const originalEnvs: Record<string, string> = {};
+
+  beforeEach(() => {
+    for (const env of ENVS_TO_RESET) {
+      if (env in process.env) {
+        originalEnvs[env] = process.env[env] || '';
+        delete process.env[env];
+      }
+    }
+  });
+
+  afterEach(() => {
+    for (const env of ENVS_TO_RESET) {
+      delete process.env[env];
+    }
+
+    for (const env of Object.keys(originalEnvs)) {
+      process.env[env] = originalEnvs[env];
+    }
+
+    jest.resetAllMocks();
+  });
+
+  let MOCKS: Record<string, object>;
+
+  const createMock = (overrides = {}) => {
+    return {
+      file: resolve(REPO_ROOT, MOCK_TEST_PATH),
+      title: 'A Test',
+      suiteTag: MOCK_TEST_PATH,
+      ...overrides,
+    };
+  };
+
+  const runLifecycleWithMocks = async (mocks: object[], fn: (objs: any) => any = () => {}) => {
+    const lifecycle = new Lifecycle();
+    const suiteTracker = SuiteTracker.startTracking(
+      lifecycle,
+      resolve(REPO_ROOT, MOCK_CONFIG_PATH)
+    );
+
+    const ret = { lifecycle, suiteTracker };
+
+    for (const mock of mocks) {
+      await lifecycle.beforeTestSuite.trigger(mock);
+    }
+
+    if (fn) {
+      fn(ret);
+    }
+
+    for (const mock of mocks.reverse()) {
+      await lifecycle.afterTestSuite.trigger(mock);
+    }
+
+    return ret;
+  };
+
+  beforeEach(() => {
+    MOCKS = {
+      WITH_TESTS: createMock({ tests: [{}] }), // i.e. a describe with tests in it
+      WITHOUT_TESTS: createMock(), // i.e. a describe with only other describes in it
+    };
+  });
+
+  it('collects metadata for a single suite with multiple describe()s', async () => {
+    const { suiteTracker } = await runLifecycleWithMocks([MOCKS.WITHOUT_TESTS, MOCKS.WITH_TESTS]);
+
+    const suites = suiteTracker.getAllFinishedSuites();
+    expect(suites.length).toBe(1);
+    const suite = suites[0];
+
+    expect(suite).toMatchObject({
+      config: MOCK_CONFIG_PATH,
+      file: MOCK_TEST_PATH,
+      tag: MOCK_TEST_PATH,
+      leafSuite: true,
+      success: true,
+    });
+  });
+
+  it('writes metadata to a file when cleanup is triggered', async () => {
+    const { lifecycle, suiteTracker } = await runLifecycleWithMocks([MOCKS.WITH_TESTS]);
+    await lifecycle.cleanup.trigger();
+
+    const suites = suiteTracker.getAllFinishedSuites();
+
+    const call = (fs.writeFileSync as jest.Mock).mock.calls[0];
+    expect(call[0]).toEqual(DEFAULT_TEST_METADATA_PATH);
+    expect(call[1]).toEqual(JSON.stringify(suites, null, 2));
+  });
+
+  it('respects TEST_METADATA_PATH env var for metadata target override', async () => {
+    process.env.TEST_METADATA_PATH = '/dev/null/fake-test-path';
+    const { lifecycle } = await runLifecycleWithMocks([MOCKS.WITH_TESTS]);
+    await lifecycle.cleanup.trigger();
+
+    expect((fs.writeFileSync as jest.Mock).mock.calls[0][0]).toEqual(
+      process.env.TEST_METADATA_PATH
```

Comment:
This is going to fail on windows. What I normally do is:

```ts
import { resolve } from 'path';
import { REPO_ROOT, createAbsolutePathSerializer } from '@kbn/dev-utils';

expect.addSnapshotSerializer(createAbsolutePathSerializer());

it('respects TEST_METADATA_PATH env var for metadata target override', async () => {
  process.env.TEST_METADATA_PATH = resolve(REPO_ROOT, 'fake-test-path');
  const { lifecycle } = await runLifecycleWithMocks([MOCKS.WITH_TESTS]);
  await lifecycle.cleanup.trigger();

  expect(fs.writeFileSync).toMatchInlineSnapshot();
});
```

Reply:
I guess you could just update the assignment to be `process.env.TEST_METADATA_PATH = resolve(REPO_ROOT, '../fake-test-path');`